app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -179,6 +207,49 @@ class WallpapersUseCases(
         }
     }
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal class DefaultInitializeWallpaperUseCase(
+        private val store: AppStore,
+        private val downloader: WallpaperDownloader,
+        private val fileManager: WallpaperFileManager,
+        private val metadataFetcher: WallpaperMetadataFetcher,
+        private val settings: Settings,
+        private val currentLocale: String,
+    ) : InitializeWallpapersUseCase {
+        override suspend fun invoke() {
+            val currentWallpaperName = withContext(Dispatchers.IO) { settings.currentWallpaper }
```

Comment:
if `settings.currentWallpaper` is only a name vs the current `Wallpaper` data object, should the name be updated to reflect that?

Reply:
Sure, that seems like a good change to me.